.github/workflows/release-please.yml

```diff
@@ -0,0 +1,81 @@
+on:
+  push:
+    branches:
+      - main
+
+name: release-please
+
+jobs:
+  release-please:
+    # The secret HCLOUD_BOT_TOKEN is only available on the main repo, not in forks.
```

Comment:
[suggestion (security)][conventional comments]: I'd like to point out that making repo-global secrets makes them accessible more broadly, than necessary. Following the principle of least privilege, I suggest restricting the access to just this job.
This can be implemented through [GitHub Environments]:
1) Make a new `hcloud-deployment` env in the repo settings.
2) Add the `HCLOUD_BOT_TOKEN` into the environment, remove it from the repo-global location.
3) [optionally?] Set up "Required reviewers" for the environment — this makes it so the job doesn't start until a human doesn't click the UI approval button. This further restricts the possibility of token leaks.
4) Add the following to this job definition:
   ```yaml
   environment:
     name: hcloud-deployment
     url: https://the-target.release.url
   ```
   (this could be just `environment: hcloud-deployment` if the URL is unknown)

[conventional comments]: https://conventionalcomments.org/
[GitHub Environments]: https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-secrets

Reply:
P.S. Moving to environments is still useful w/o the required reviewers bit.

Reply:
Also a useful suggestion, I think we will do this once we are using Terraform to manage our GitHub repo settings, to avoid too much toil in configuring this across all X repos.